### DISCUSSION TITLE
Using proxyWithComputed on vanilla js app

### DISCUSSION THREAD

**Discussion by wobsoriano** [pmndrs/valtio#356] [Q&A]
Hi, was trying to import `proxyWithComputed` on a vanilla js app with vite:

```ts
import { proxyWithComputed } from "valtio/utils";

const state2 = proxyWithComputed(
  {
    count: 1,
  },
  {
    doubled: (snap) => snap.count * 2,
  }
);
```

but this gives me an error:

```
[plugin:vite:import-analysis] Failed to resolve import "react" from "node_modules/valtio/esm/utils.js". Does the file exist?
..[/valtio/esm/index.js:1:0]()
10 |  
11 |  // node_modules/valtio/esm/index.js
12 |  import { useRef, useState, useEffect, useReducer, useCallback, useMemo, useLayoutEffect, useDebugValue } from "react";
   |                                                                                                                 ^
13 |  var isSSR = typeof window === "undefined" || !window.navigator || /ServerSideRendering|^Deno\//.test(window.navigator.userAgent);
14 |
```

Can `proxyWithComputed` be used without react?  I'm using version 1.2.12

**Comment by dai-shi** [comment]
It's expected to work without react. can be a bundling issue.
But, I wonder why vite is trying to read `/valtio/esm/index.js` file.
Do you have `import { ... } from 'valtio' somewhere in your code?

**Reply by wobsoriano** [reply]
I only have this currently:

```ts
import { proxy } from "valtio/vanilla";
import { proxyWithComputed } from "valtio/utils";
```

Using the latest version of vite too.

**Reply by dai-shi** [reply]
It looks good.. You might need to create a small reproduction to further investigate.
But, before that, can you quickly try #338?
https://ci.codesandbox.io/status/pmndrs/valtio/pr/338
Find "Local Install Instructions" ☝️ 

Also mentioning @Aslemammad if they have some hints.

**Reply by wobsoriano** [reply]
Here's a basic repro https://codesandbox.io/s/peaceful-worker-8t2tb?file=/src/App.tsx

**Reply by wobsoriano** [reply]
Not sure but, you think it's related to `valtio` being imported in `valtio/esm/utils.js`?

```js
import { subscribe, snapshot, proxy, getVersion, ref } from 'valtio/vanilla';
import { createProxy, isChanged } from 'proxy-compare';
import { proxy as proxy$1 } from 'valtio';
```

UPDATE: I tried manually editing `node_modules/valtio/esm/utils.js` and it's now working

```js
import { subscribe, snapshot, proxy, getVersion, ref } from 'valtio/vanilla';
import { createProxy, isChanged } from 'proxy-compare';
// import { proxy as proxy$1 } from 'valtio';
import { proxy as proxy$1 } from 'valtio/vanilla'; //This is the culprit
```

**Reply by dai-shi** [reply]
Nice catch!!!
Now, I understand the issue. I will open a PR to fix it.

**Reply by dai-shi** [reply]
Can you try #358?
https://ci.codesandbox.io/status/pmndrs/valtio/pr/358
See "Local Install Instructions" ☝️

**Reply by wobsoriano** [reply]
Done. Working now!

**Reply by dai-shi** [reply]
Thanks. I will cut a release in a couple of days. Feel free to keep using the codesandbox build meanwhile.
